### ISSUE TITLE
Make filtering for tags more refined

### ISSUE THREAD

**Issue by simonpoole** [mapbox/osmcha-frontend#184]
Enhancement request

https://osmcha.mapbox.com/changesets/51496339?filters=%7B%22ids%22%3A%5B%7B%22label%22%3A%2251496339%22%2C%22value%22%3A%2251496339%22%7D%5D%7D

lit up like a Christmas tree, however the current tag filter simply shows all tagged changesets, which makes if difficult to detect the high-priority "must fix now" changesets because of all the noise, which is why I assume it took a day for it to be noticed.

Simply allow to show only changesets with more than a selectable number of tags would seem to go a long way to making real issues stand out.

**Comment by kepta** [comment]
If I understand it correctly,

1. There needs to be a button, which should be prominent.
2. This button can be worded something like `Show Me Suspicious`.
3. This button would simply apply filters which will flag the important ones.

Use case:
This will help new comers (to OSMCha), by simply clicking a button they can see changesets which are suspicious. 

Extension:
We can discuss of having a simple modifier which also automatically applies the users current geo-position. 

In short an `automagic` button which takes care of all the filters and shows relevant changesets.

**Comment by simonpoole** [comment]
I would not necessarily see a need for something automagic, just a filter that allows you to only see say changesets that have at least 3 tags and more. That should really be simple enough for all including beginners to understand, but naturally it could be made more luxurious if somebody has the time.

**Comment by planemad** [comment]
> Simply allow to show only changesets with more than a selectable number of tags would seem to go a long way to making real issues stand out.

@simonpoole Do you mean something like "show me latest unreviewed changesets with the maximum number of suspicious reasons"? It would help if you can clarify exactly what you have in mind.

**Comment by batpad** [comment]
@simonpoole thanks for the ticket!

So I think there's probably separate threads here on making things easier, etc. but in terms of features, I see two things concretely:

1> Be able to filter changesets by "number of suspicions" - so you'd be able to say "only show me changesets that have been tagged with more than 3 reasons", for eg. - @willemarcel does this seem like something easy / doable in the API?

2> In a medium / longer term, I think it maybe nice to have a notion of "priority" that's some-how set by the flaggers. I think there's been some thinking around this, but it's not concrete yet - would love to think about how that could work, but I think we should be able to implement 1> pretty quickly.

**Comment by simonpoole** [comment]
@planemad minimum number of tags not maximum, while theoretically you could express that with the current boolean expressions, it is obviously not practical.

@batpad yes the alternative would be, as most spam filters do, generate a score (associate a score with each of the tags and add them all up) and then allow the user to filter on a minimum score (essentially what I'm suggesting is simply a score of 1 per tag)..

**Comment by willemarcel** [comment]
@batpad yes, I'll work on it!

**Comment by batpad** [comment]
> @batpad yes the alternative would be, as most spam filters do, generate a score (associate a score with each of the tags and add them all up) and then allow the user to filter on a minimum score (essentially what I'm suggesting is simply a score of 1 per tag)..

Yes, I think this would be a good longer-term goal -- I think, though, our current and past attempts at doing something like coming up with an automated "severity score" or so have failed due to, really, the diversity of possible edits in OSM, good and bad, and it being really hard to automagically assess severity. I think the current idea has been more along the lines of "let people figure out what kind of reasons are important to them and filter accordingly", and I think getting into attaching "score" values to types of edits would ideally be a much broader community project than some algorithm we define that decides that :-) - would love to help if that's a broader discussion that's good to start somewhere.

For now, I really like the idea of being able  to search by minimum number of reason tags, thanks for taking this on @willemarcel and thanks for the idea @simonpoole :-)

**Comment by willemarcel** [comment]
@batpad I just finished to add this filter to the API https://github.com/willemarcel/osmcha-django/pull/127
